src/v/archival/ntp_archiver_service.cc

```diff
@@ -3257,6 +3261,11 @@ const storage::ntp_config& ntp_archiver::ntp_config() const {
     return _parent.log()->config();
 }
 
+const cloud_storage::remote_path_provider&
+ntp_archiver::remote_path_provider() const {
+    return _parent.archival_meta_stm()->path_provider();
```

Comment:
Same comment as before, it's always the same path provider for the partition and it lives in archival STM so why not connect manifest and path provider on that level instead of doing this in every call? This makes code confusing because the reader might think that the path provider could be changed from upload to upload. Also, the `async_manifest_view` is getting the path provider in c-tor and `partition_manifest` doesn't. Which is also a bit confusing and not uniform.

Reply:
Pasting from other PR comments:

I think the shift in mindset is that the manifest view and the partition manifest are not equivalent to each other. The manifest view is much more full-featured and must be able to access paths of an STM manifest and spillover manifests, unlike the partition_manifest class which is focused on tracking member fields and serialization.